src/components/api/json-rpc/che-json-rpc-master-api.ts

```diff
@@ -126,8 +134,13 @@ export class CheJsonRpcMasterApi {
   }
 
   reconnect(entrypoint: string): void {
-    this.$log.warn('WebSocket connection is closed.');
-    if (this.reconnectionAttemptNumber === this.maxReconnectionAttempts) {
+    if (this.reconnectionAttemptNumber < this.maxReconnectionAttempts) {
+      this.$log.warn('WebSocket connection is closed.');
+    } else {
+      this.globalWarningBannerService.addMessage(`WebSocket connections "${entrypoint}" are failing due to network restrictions.
+       ${this.branding.getName()} workspaces may not be usable. Please refer to the
+       <a href="${this.branding.getDocs().webSocketTroubleshooting}" target="_blank">Network Troubleshooting</a>
```

Comment:
Do we have an issue to make sure that this article will appear in CRW and the right reference will be filled in in CRW product.json?

Reply:
No. We don't have an issue to make sure that this article will appear in CRW and the right reference will be filled in in CRW product.json